src/jit/gentree.h

```diff
@@ -2080,31 +2080,61 @@ class GenTreeUseEdgeIterator final
     friend GenTreeUseEdgeIterator GenTree::UseEdgesBegin();
     friend GenTreeUseEdgeIterator GenTree::UseEdgesEnd();
 
+    enum
+    {
```

Comment:
I think it would be useful to add a bit more commenting to the GenTreeUseEdgeIterator header comment, discussing how this is hot (maybe why it is hot), and how that informs the design by pushing things to the constructor, limiting work in the "advance" cases, etc., etc.